src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -8114,6 +8104,37 @@ namespace VariableSpeedCoils {
         return Speeds;
     }
 
+    Real64 GetVSCoilRatedSourceTemp(EnergyPlusData &state,
+                                    int const &CoilIndex, // index to cooling coil
+                                    bool &ErrorsFound     // set to true if problem
+    )
+    {
+        switch (state.dataVariableSpeedCoils->VarSpeedCoil(CoilIndex).VSCoilType) {
+        case DataHVACGlobals::Coil_CoolingWaterToAirHPVSEquationFit: {
+            return RatedInletWaterTemp;
+        }
+        case DataHVACGlobals::Coil_HeatingWaterToAirHPVSEquationFit: {
+            return RatedInletWaterTempHeat;
+        }
+        case DataHVACGlobals::CoilDX_HeatPumpWaterHeaterVariableSpeed: {
+            return state.dataVariableSpeedCoils->VarSpeedCoil(CoilIndex).WHRatedInletWaterTemp;
+        }
+        case DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed: {
+            return RatedAmbAirTemp;
+        }
+        case DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed: {
+            return RatedAmbAirTempHeat;
+        }
+        default: {
+            ShowSevereError(state,
+                            "GetVSCoilRatedSourceTemp: Could not find rated source temperature, Type= VS DX Coil Name=\"" +
+                                state.dataVariableSpeedCoils->VarSpeedCoil(CoilIndex).Name + "\"");
+            ErrorsFound = true;
+            return 0.0;
+        }
+        }
+    }
+
```

Comment:
New function that determine which rated source temperature to use.

Reply:
This warning will never get used, correct? Better to assert(false) here?